envoyauth/request.go

```diff
@@ -56,6 +67,13 @@ func RequestToInput(req any, logger logging.Logger, protoSet *protoregistry.File
 		body = req.GetAttributes().GetRequest().GetHttp().GetBody()
 		headers = req.GetAttributes().GetRequest().GetHttp().GetHeaders()
 		rawBody = req.GetAttributes().GetRequest().GetHttp().GetRawBody()
+
+		if req.GetAttributes().GetSource().GetPrincipal() != "" {
+			sourcePrincipal, err = ast.InterfaceToValue(req.GetAttributes().GetSource().GetPrincipal())
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Nit: we could pull out the prinicpal to a var to not do the same work twice.

```suggestion
		if p := req.GetAttributes().GetSource().GetPrincipal(); p != "" {
			sourcePrincipal, err = ast.InterfaceToValue(p)
			if err != nil {
				return nil, err
			}
		}
```

Reply:
Added!